pkg/cluster/internal/providers/docker/node.go

```diff
@@ -100,34 +100,19 @@ type nodeCmd struct {
 }
 
 func (c *nodeCmd) Run() error {
-	args := []string{
-		"exec",
-		// run with privileges so we can remount etc..
-		// this might not make sense in the most general sense, but it is
-		// important to many kind commands
-		"--privileged",
-	}
-	if c.stdin != nil {
-		args = append(args,
-			"-i", // interactive so we can supply input
-		)
-	}
-	// set env
-	for _, env := range c.env {
-		args = append(args, "-e", env)
+	var args []string
+	switch c.command {
```

Comment:
This seems like a fragile hack. The cp replacement for example is *not* identical to running posix cp commands on the nodes. I don't think we should magically switch to a different mode of operation based on argv[0].

If we do go forward with this change probably it should call some new node methods.

Reply:
Totally agree.  I was hoping to tease the approach and start the conversation before going through and introducing new methods on the Note interface, something that I am pretty sure I'd need some direction from ya'll on.